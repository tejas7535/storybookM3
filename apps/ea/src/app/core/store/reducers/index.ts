import { environment } from '@ea/environments/environment';
import {
  ActionReducerMap,
  createFeatureSelector,
  MetaReducer,
} from '@ngrx/store';

import {
  CalculationParametersState,
  CatalogCalculationResultState,
  CO2UpstreamCalculationResultState,
  FrictionCalculationResultState,
  ProductSelectionState,
  SettingsState,
} from '../models';
import { calculationParametersReducer } from './calculation-parameters/calculation-parameters.reducer';
import { catalogCalculationResultReducer } from './calculation-result/catalog-calculation-result.reducer';
import { co2UpstreamCalculationResultReducer } from './calculation-result/co2-upstream-calculation-result.reducer';
import { frictionCalculationResultReducer } from './calculation-result/friction-calculation-result.reducer';
import { productSelectionReducer } from './product-selection/product-selection.reducer';
import { settingsReducer } from './settings/settings.reducer';

export interface AppState {
  calculationParameters: CalculationParametersState;
  productSelection: ProductSelectionState;
  settings: SettingsState;
  frictionCalculationResult: FrictionCalculationResultState;
  co2UpstreamCalculationResult: CO2UpstreamCalculationResultState;
  catalogCalculationResult: CatalogCalculationResultState;
}

export const reducers: ActionReducerMap<AppState> = {
  calculationParameters: calculationParametersReducer,
  productSelection: productSelectionReducer,
  settings: settingsReducer,
  frictionCalculationResult: frictionCalculationResultReducer,
  co2UpstreamCalculationResult: co2UpstreamCalculationResultReducer,
  catalogCalculationResult: catalogCalculationResultReducer,
};

export const metaReducers: MetaReducer<AppState>[] = !environment.production
  ? []
  : /* istanbul ignore next: very difficult */ [];

export const getCalculationParametersState =
  createFeatureSelector<CalculationParametersState>('calculationParameters');

export const getProductSelectionState =
  createFeatureSelector<ProductSelectionState>('productSelection');

export const getSettingsState =
  createFeatureSelector<SettingsState>('settings');

export const getFrictionCalculationResultState =
  createFeatureSelector<FrictionCalculationResultState>(
    'frictionCalculationResult'
  );

export const getCO2UpstreamCalculationResultState =
  createFeatureSelector<CO2UpstreamCalculationResultState>(
    'co2UpstreamCalculationResult'
  );

export const getCatalogCalculationResultState =
  createFeatureSelector<CatalogCalculationResultState>(
    'catalogCalculationResult'
  );
