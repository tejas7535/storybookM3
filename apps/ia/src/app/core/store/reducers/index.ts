import { InjectionToken } from '@angular/core';
import { Params, RouterStateSnapshot } from '@angular/router';

import * as fromRouter from '@ngrx/router-store';
import {
  ActionReducerMap,
  createFeatureSelector,
  MetaReducer,
} from '@ngrx/store';

import { environment } from '../../../../environments/environment';
import * as fromFilter from './filter/filter.reducer';

export interface RouterStateUrl {
  url: string;
  queryParams: Params;
  params: Params;
}

export interface AppState {
  router: fromRouter.RouterReducerState<RouterStateUrl>;
  [fromFilter.filterKey]: fromFilter.FilterState;
}

export const ROOT_REDUCERS = new InjectionToken<ActionReducerMap<AppState>>(
  'Root reducers token',
  {
    factory: () => ({
      router: fromRouter.routerReducer,
      [fromFilter.filterKey]: fromFilter.filterReducer,
    }),
  }
);

export const metaReducers: MetaReducer<AppState>[] = environment.production
  ? []
  : [];

export const selectRouterState =
  createFeatureSelector<fromRouter.RouterReducerState<RouterStateUrl>>(
    'router'
  );

export const selectFilterState = createFeatureSelector<fromFilter.FilterState>(
  fromFilter.filterKey
);

export class CustomSerializer
  implements fromRouter.RouterStateSerializer<RouterStateUrl>
{
  /**
   * Serialize the router state
   */
  public serialize(routerState: RouterStateSnapshot): RouterStateUrl {
    let route = routerState.root;

    while (route.firstChild) {
      route = route.firstChild;
    }

    const {
      url,
      root: { queryParams },
    } = routerState;
    const { params } = route;

    // Only return an object including the URL, params and query params
    // instead of the entire snapshot
    return { url, params, queryParams };
  }
}
