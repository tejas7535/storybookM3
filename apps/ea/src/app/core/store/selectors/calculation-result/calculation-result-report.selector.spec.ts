import {
  APP_STATE_MOCK,
  CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
} from '@ea/testing/mocks';
import { DOWNSTREAM_STATE_MOCK } from '@ea/testing/mocks/store/downstream-calculation-state.mock';

import { AppState } from '../../reducers';
import {
  combineLoadcaseResultItemValuesByKey,
  getAllErrors,
  getCalculationsWithResult,
  getCO2EmissionReport,
  getFrictionalalPowerlossReport,
  getLubricationReport,
  getOverrollingFrequencies,
  getRatingLifeResultReport,
  getReportDownstreamErrors,
  getReportErrors,
  getReportNotes,
  getReportWarnings,
  getResultInput,
  getSelectedCalculations,
  pdfReportAvailable,
} from './calculation-result-report.selector';

describe('Calculation Result Selector', () => {
  const mockState: AppState = {
    ...APP_STATE_MOCK,
    co2UpstreamCalculationResult: {
      calculationResult: {
        unit: 'kg',
        upstreamEmissionFactor: 3,
        upstreamEmissionTotal: 5,
        weight: 0.5,
      },
      isLoading: false,
    },
    catalogCalculationResult: {
      result: {
        reportInputSuborinates: {
          inputSubordinates: [
            {
              hasNestedStructure: true,
              title: 'some title',
            },
          ],
        },
        reportMessages: {
          errors: ['Errors'],
          warnings: ['Warnings'],
          notes: ['Notes'],
        },
      },
      isLoading: false,
    },
  };

  describe('combineLoadcaseResultItemValuesByKey', () => {
    it('should return loadcase values for specified key', () => {
      const key = 'test';
      const mockResultItems = [
        {
          test: {
            value: 1,
            unit: 'unit',
            short: 'short',
            loadcaseName: 'loadcase1',
            title: 'test',
          },
          ignore: {
            value: -1,
            unit: 'unit',
            short: 'short',
            loadcaseName: 'loadcase1',
            title: 'ignore',
          },
        },
        {
          test: {
            value: 2,
            unit: 'unit',
            short: 'short',
            loadcaseName: 'loadcase2',
            title: 'test',
          },
          ignore: {
            value: -1,
            unit: 'unit',
            short: 'short',
            loadcaseName: 'loadcase2',
            title: 'ignore',
          },
        },
      ];

      const expected = [
        {
          value: 1,
          loadcaseName: 'loadcase1',
        },
        {
          value: 2,
          loadcaseName: 'loadcase2',
        },
      ];

      const result = combineLoadcaseResultItemValuesByKey(mockResultItems, key);

      expect(result).toEqual(expected);
    });
  });

  describe('emptry result', () => {
    describe('getCO2EmissionReport', () => {
      it('should return the co2 emission report data', () => {
        expect(getCO2EmissionReport(mockState)).toMatchSnapshot();
      });
    });

    describe('getCalculationWithResult', () => {
      it('should return the calculations with result', () => {
        expect(getCalculationsWithResult(mockState)).toMatchSnapshot();
      });
    });

    describe('getSelectedCalculations', () => {
      it('should return the selected calculations', () => {
        expect(getSelectedCalculations(mockState)).toMatchSnapshot();
      });
    });

    describe('getResultInput', () => {
      it('should return result input', () => {
        expect(getResultInput(mockState)).toMatchSnapshot();
      });
    });

    describe('getReportErrors', () => {
      it('should return report errors', () => {
        expect(getReportErrors(mockState)).toMatchSnapshot();
      });
    });

    describe('getReportDownstreamErrors', () => {
      it('should return report downstream errors', () => {
        expect(getReportDownstreamErrors(mockState)).toMatchSnapshot();
      });
    });

    describe('getReportWarnings', () => {
      it('should return report warnings', () => {
        expect(getReportWarnings(mockState)).toMatchSnapshot();
      });
    });

    describe('getReportNotes', () => {
      it('should return report notes', () => {
        expect(getReportNotes(mockState)).toMatchSnapshot();
      });
    });
  });

  describe('full result', () => {
    describe('getCO2EmissionReport', () => {
      it('should return the co2 emission report data', () => {
        expect(
          getCO2EmissionReport({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getCalculationWithResult', () => {
      it('should return the calculations with result', () => {
        expect(
          getCalculationsWithResult({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getSelectedCalculations', () => {
      it('should return the selected calculations', () => {
        expect(
          getSelectedCalculations({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getFrictionalPowerlossReport', () => {
      it('should return frictionalPowerloss', () => {
        expect(
          getFrictionalalPowerlossReport({
            ...mockState,
            downstreamCalculationState: {
              ...DOWNSTREAM_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });

      it('should return null result if no value frictionalPowerloss', () => {
        expect(
          getFrictionalalPowerlossReport({
            ...mockState,
            downstreamCalculationState: {},
          })
        ).toMatchSnapshot();
      });
    });

    describe('getLubricationReport', () => {
      it('should return lubrication', () => {
        expect(
          getLubricationReport({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getRatingLifeResultReport', () => {
      it('should return ratingLife', () => {
        expect(
          getRatingLifeResultReport({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getOverrollingFrequencies', () => {
      it('should return overrollingFrequencies', () => {
        expect(
          getOverrollingFrequencies({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getResultInput', () => {
      it('should return result input', () => {
        expect(
          getResultInput({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getReportErrors', () => {
      it('should return report errors', () => {
        expect(
          getReportErrors({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getAllErrors', () => {
      it('should return report errors', () => {
        expect(
          getAllErrors({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getReportWarnings', () => {
      it('should return report warnings', () => {
        expect(
          getReportWarnings({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });

    describe('getReportNotes', () => {
      it('should return report notes', () => {
        expect(
          getReportNotes({
            ...mockState,
            catalogCalculationResult: {
              ...CATALOG_CALCULATION_FULL_RESULT_STATE_MOCK,
            },
          })
        ).toMatchSnapshot();
      });
    });
  });

  describe('pdfReportAvailable', () => {
    it('should be truthy if something is selected', () => {
      expect(pdfReportAvailable(mockState)).toBeTruthy();
    });
  });
});
