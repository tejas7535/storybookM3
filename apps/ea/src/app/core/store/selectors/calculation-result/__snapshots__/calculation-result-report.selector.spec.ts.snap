// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculation Result Selector emptry result getCO2EmissionReport should return the co2 emission report data 1`] = `
Object {
  "co2_downstream": Object {
    "emission": 10.296,
    "emissionPercentage": 67.31171548117155,
    "loadcases": Array [
      Object {
        "emission": 2.848,
        "emissionPercentage": 18.619246861924683,
        "id": "Loadcase 1",
        "operatingTimeInHours": 3506,
        "unit": "kg",
      },
      Object {
        "emission": 7.4478,
        "emissionPercentage": 48.69116108786611,
        "id": "Loadcase 2",
        "operatingTimeInHours": 5260,
        "unit": "kg",
      },
    ],
  },
  "co2_upstream": 5,
  "co2_upstreamEmissionPercentage": 32.68828451882845,
  "totalEmission": 15.296,
}
`;

exports[`Calculation Result Selector emptry result getCalculationWithResult should return the calculations with result 1`] = `
Array [
  Object {
    "disabled": true,
    "label": "calculationTypes.co2ForProduction",
    "name": "emission",
    "resultAvailable": true,
    "selected": true,
    "svgIcon": "co2",
    "visible": true,
  },
  Object {
    "disabled": false,
    "icon": "compress",
    "label": "calculationTypes.frictionalPowerloss",
    "name": "frictionalPowerloss",
    "resultAvailable": true,
    "selected": true,
    "visible": true,
  },
]
`;

exports[`Calculation Result Selector emptry result getReportDownstreamErrors should return report downstream errors 1`] = `Array []`;

exports[`Calculation Result Selector emptry result getReportErrors should return report errors 1`] = `
Array [
  "Errors",
]
`;

exports[`Calculation Result Selector emptry result getReportNotes should return report notes 1`] = `
Array [
  "Notes",
]
`;

exports[`Calculation Result Selector emptry result getReportWarnings should return report warnings 1`] = `
Array [
  "Warnings",
]
`;

exports[`Calculation Result Selector emptry result getResultInput should return result input 1`] = `
Array [
  Object {
    "hasNestedStructure": true,
    "title": "some title",
  },
]
`;

exports[`Calculation Result Selector emptry result getSelectedCalculations should return the selected calculations 1`] = `
Array [
  Object {
    "disabled": true,
    "label": "calculationTypes.co2ForProduction",
    "name": "emission",
    "resultAvailable": true,
    "selected": true,
    "svgIcon": "co2",
    "visible": true,
  },
  Object {
    "disabled": false,
    "icon": "compress",
    "label": "calculationTypes.frictionalPowerloss",
    "name": "frictionalPowerloss",
    "resultAvailable": true,
    "selected": true,
    "visible": true,
  },
]
`;

exports[`Calculation Result Selector full result getAllErrors should return report errors 1`] = `
Array [
  "Calculation failed for some of the reason",
]
`;

exports[`Calculation Result Selector full result getCO2EmissionReport should return the co2 emission report data 1`] = `
Object {
  "co2_downstream": Object {
    "emission": 10.296,
    "emissionPercentage": 67.31171548117155,
    "loadcases": Array [
      Object {
        "emission": 2.848,
        "emissionPercentage": 18.619246861924683,
        "id": "Loadcase 1",
        "operatingTimeInHours": 3506,
        "unit": "kg",
      },
      Object {
        "emission": 7.4478,
        "emissionPercentage": 48.69116108786611,
        "id": "Loadcase 2",
        "operatingTimeInHours": 5260,
        "unit": "kg",
      },
    ],
  },
  "co2_upstream": 5,
  "co2_upstreamEmissionPercentage": 32.68828451882845,
  "totalEmission": 15.296,
}
`;

exports[`Calculation Result Selector full result getCalculationWithResult should return the calculations with result 1`] = `
Array [
  Object {
    "disabled": true,
    "label": "calculationTypes.co2ForProduction",
    "name": "emission",
    "resultAvailable": true,
    "selected": true,
    "svgIcon": "co2",
    "visible": true,
  },
  Object {
    "disabled": false,
    "icon": "compress",
    "label": "calculationTypes.frictionalPowerloss",
    "name": "frictionalPowerloss",
    "resultAvailable": true,
    "selected": true,
    "visible": true,
  },
]
`;

exports[`Calculation Result Selector full result getFrictionalPowerlossReport should return frictionalPowerloss 1`] = `
Array [
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 3,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 5,
      },
    ],
    "short": "NR",
    "title": "totalFrictionalPowerLoss",
    "unit": "W",
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 0.03,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 0.041,
      },
    ],
    "short": "MR",
    "title": "totalFrictionalTorque",
    "unit": "N m",
  },
]
`;

exports[`Calculation Result Selector full result getFrictionalPowerlossReport should return null result if no value frictionalPowerloss 1`] = `null`;

exports[`Calculation Result Selector full result getLubricationReport should return lubrication 1`] = `
Array [
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 28,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 65.4,
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": 65.4,
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": 65.4,
      },
    ],
    "short": "ny",
    "title": "operatingViscosity",
    "unit": "mm²/s",
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 97.9,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 45.8,
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": 45.8,
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": 45.8,
      },
    ],
    "short": "ny1",
    "title": "referenceViscosity",
    "unit": "mm²/s",
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 0.29,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 1.43,
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": 1.43,
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": 1.43,
      },
    ],
    "short": "kappa",
    "title": "viscosityRatio",
    "unit": undefined,
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 0.53,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 2.77,
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": 2.77,
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": 2.77,
      },
    ],
    "short": "a_ISO",
    "title": "lifeAdjustmentFactor",
    "unit": undefined,
    "warning": undefined,
  },
  Object {
    "short": "tfR",
    "title": "guideIntervalRelubrication",
    "unit": "h",
    "value": "~ 11360",
  },
]
`;

exports[`Calculation Result Selector full result getOverrollingFrequencies should return overrollingFrequencies 1`] = `
Array [
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": "18.5926",
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": "46.4816",
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": "46.4816",
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": "46.4816",
      },
    ],
    "short": "BPFI",
    "title": "BPFI",
    "unit": "1/s",
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": "11.4074",
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": "28.5184",
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": "28.5184",
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": "28.5184",
      },
    ],
    "short": "BPFO",
    "title": "BPFO",
    "unit": "1/s",
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": "6.1871",
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": "15.4676",
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": "15.4676",
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": "15.4676",
      },
    ],
    "short": "BSF",
    "title": "BSF",
    "unit": "1/s",
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": "1.2675",
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": "3.1687",
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": "3.1687",
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": "3.1687",
      },
    ],
    "short": "FTF",
    "title": "FTF",
    "unit": "1/s",
    "warning": undefined,
  },
]
`;

exports[`Calculation Result Selector full result getRatingLifeResultReport should return ratingLife 1`] = `
Array [
  Object {
    "short": "Lh10",
    "title": "lh10",
    "unit": "h",
    "value": "152277",
  },
  Object {
    "short": "Lh_nm",
    "title": "lh_nm",
    "unit": "h",
    "value": "415307",
  },
  Object {
    "short": "P",
    "title": "p",
    "unit": "N",
    "value": "1028.71",
  },
  Object {
    "short": "n",
    "title": "n",
    "unit": "1/min",
    "value": "270.0",
  },
  Object {
    "short": "S0_min",
    "title": "s0_min",
    "value": "2.844",
  },
  Object {
    "short": "P0_max",
    "title": "p0_max",
    "unit": "N",
    "value": "1125.00",
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 200,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 1125,
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": 1125,
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": 1125,
      },
    ],
    "short": "P0",
    "title": "p0",
    "unit": "N",
    "warning": undefined,
  },
  Object {
    "loadcaseValues": Array [
      Object {
        "loadcaseName": "Loadcase 1",
        "value": 200,
      },
      Object {
        "loadcaseName": "Loadcase 2",
        "value": 1250,
      },
      Object {
        "loadcaseName": "Loadcase 3",
        "value": 1250,
      },
      Object {
        "loadcaseName": "Loadcase 4",
        "value": 1250,
      },
    ],
    "short": "P_i",
    "title": "p_i",
    "unit": "N",
    "warning": undefined,
  },
]
`;

exports[`Calculation Result Selector full result getReportErrors should return report errors 1`] = `
Array [
  "Calculation failed for some of the reason",
]
`;

exports[`Calculation Result Selector full result getReportNotes should return report notes 1`] = `
Array [
  "Please note that operating temperature needs to meet conditions...",
]
`;

exports[`Calculation Result Selector full result getReportWarnings should return report warnings 1`] = `
Array [
  "The total of the load case time portions is less than 100%. The missing time portion is assumed to be a stationary period.",
  "  · Sum of time proportions = 90.000 %",
  "Please note the miscibility of greases in the chapter on grease lubrication.",
]
`;

exports[`Calculation Result Selector full result getResultInput should return result input 1`] = `
Array [
  Object {
    "hasNestedStructure": false,
    "subItems": Array [
      Object {
        "designation": "Designation",
        "hasNestedStructure": false,
        "value": "2303-TVH",
      },
      Object {
        "abbreviation": "d",
        "designation": "Inside diameter",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": "17.000",
      },
      Object {
        "abbreviation": "D",
        "designation": "Outside diameter",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": "47.000",
      },
      Object {
        "abbreviation": "B",
        "designation": "Width",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": "19.000",
      },
      Object {
        "abbreviation": "C",
        "designation": "Basic dynamic load rating",
        "hasNestedStructure": false,
        "unit": "N",
        "value": "13900",
      },
      Object {
        "abbreviation": "C0",
        "designation": "Basic static load rating",
        "hasNestedStructure": false,
        "unit": "N",
        "value": "3200",
      },
      Object {
        "abbreviation": "Cu",
        "designation": "Fatigue limit load",
        "hasNestedStructure": false,
        "unit": "N",
        "value": "203",
      },
      Object {
        "abbreviation": "n_ref",
        "designation": "Reference speed",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": "14900.0",
      },
      Object {
        "abbreviation": "n_lim",
        "designation": "Limiting speed",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": "17000.0",
      },
      Object {
        "abbreviation": "n_lim_o",
        "designation": "Limiting speed, oil",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": "17000.0",
      },
      Object {
        "abbreviation": "n_lim_g",
        "designation": "Limiting speed, grease",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": "14400.0",
      },
    ],
    "title": "Bearing",
    "titleID": "STRING_OUTP_BEARING",
  },
  Object {
    "hasNestedStructure": false,
    "meaningfulRound": true,
    "subItems": Array [
      Object {
        "abbreviation": "BPFFO",
        "designation": "Overrolling frequency factor on outer ring",
        "hasNestedStructure": false,
        "value": "3.4222",
      },
      Object {
        "abbreviation": "BPFFI",
        "designation": "Overrolling frequency factor on inner ring",
        "hasNestedStructure": false,
        "value": "5.5778",
      },
      Object {
        "abbreviation": "BSFF",
        "designation": "Overrolling frequency factor on rolling element",
        "hasNestedStructure": false,
        "value": "1.8561",
      },
      Object {
        "abbreviation": "RPFFB",
        "designation": "Ring pass frequency factor on rolling element",
        "hasNestedStructure": false,
        "value": "3.7122",
      },
      Object {
        "abbreviation": "FTFF_i",
        "designation": "Speed factor of rolling element set for rotating inner ring",
        "hasNestedStructure": false,
        "value": "0.3802",
      },
      Object {
        "abbreviation": "FTFF_o",
        "designation": "Speed factor of rolling element set for rotating outer ring",
        "hasNestedStructure": false,
        "value": "0.6198",
      },
    ],
    "title": "Basic frequency factors related to 1/s",
    "titleID": "STRING_OUTP_BASIC_FREQUENCIES_FACTOR",
  },
  Object {
    "hasNestedStructure": false,
    "subItems": Array [
      Object {
        "designation": "Permitted lubricants",
        "hasNestedStructure": false,
        "value": "Oil or grease",
      },
      Object {
        "designation": "Type of lubrication",
        "hasNestedStructure": false,
        "value": "grease",
      },
      Object {
        "designation": "Type of grease",
        "hasNestedStructure": false,
        "value": "Arcanol MULTI2",
      },
      Object {
        "abbreviation": "ny 40",
        "designation": "Viscosity at 40°C",
        "hasNestedStructure": false,
        "unit": "mm²/s",
        "value": "110.0",
      },
      Object {
        "abbreviation": "ny 100",
        "designation": "Viscosity at 100°C",
        "hasNestedStructure": false,
        "unit": "mm²/s",
        "value": "11.0",
      },
      Object {
        "designation": "Contamination",
        "hasNestedStructure": false,
        "value": "normal cleanliness",
      },
      Object {
        "abbreviation": "dQ/dt",
        "designation": "External heat flow",
        "hasNestedStructure": false,
        "unit": "kW",
        "value": "0.0",
      },
    ],
    "title": "Lubrication data",
    "titleID": "STRING_OUTP_LUBRICATING_DATA",
  },
  Object {
    "hasNestedStructure": false,
    "subItems": Array [
      Object {
        "abbreviation": "t",
        "designation": "Ambient temperature",
        "hasNestedStructure": false,
        "unit": "°C",
        "value": "20.0",
      },
      Object {
        "designation": "Environmental influence",
        "hasNestedStructure": false,
        "value": "0.8 (moderate)",
      },
      Object {
        "designation": "Requisite reliability",
        "hasNestedStructure": false,
        "value": "90 %",
      },
      Object {
        "designation": "Condition of rotation",
        "hasNestedStructure": false,
        "value": "rotating inner ring",
      },
    ],
    "title": "Other conditions",
    "titleID": "STRING_OUTP_MISCELLANEOUS_DATA",
  },
  Object {
    "hasNestedStructure": true,
    "subItems": Array [
      Object {
        "hasNestedStructure": false,
        "subItems": Array [
          Object {
            "designation": "Time portion (q)",
            "hasNestedStructure": false,
            "unit": "%",
            "value": "60.000",
          },
          Object {
            "designation": "Type of motion",
            "hasNestedStructure": false,
            "value": "rotating",
          },
          Object {
            "designation": "Speed (n_i)",
            "hasNestedStructure": false,
            "unit": "1/min",
            "value": "200.00",
          },
          Object {
            "designation": "Mean operating temperature (T)",
            "hasNestedStructure": false,
            "unit": "°C",
            "value": "70",
          },
          Object {
            "designation": "Radial load (Fr)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "200.0",
          },
          Object {
            "designation": "Axial load (Fa)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "0.0",
          },
        ],
        "title": "Loadcase 1",
      },
      Object {
        "hasNestedStructure": false,
        "subItems": Array [
          Object {
            "designation": "Time portion (q)",
            "hasNestedStructure": false,
            "unit": "%",
            "value": "10.000",
          },
          Object {
            "designation": "Type of motion",
            "hasNestedStructure": false,
            "value": "rotating",
          },
          Object {
            "designation": "Speed (n_i)",
            "hasNestedStructure": false,
            "unit": "1/min",
            "value": "500.00",
          },
          Object {
            "designation": "Mean operating temperature (T)",
            "hasNestedStructure": false,
            "unit": "°C",
            "value": "50",
          },
          Object {
            "designation": "Radial load (Fr)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "500.0",
          },
          Object {
            "designation": "Axial load (Fa)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "500.0",
          },
        ],
        "title": "Loadcase 2",
      },
      Object {
        "hasNestedStructure": false,
        "subItems": Array [
          Object {
            "designation": "Time portion (q)",
            "hasNestedStructure": false,
            "unit": "%",
            "value": "10.000",
          },
          Object {
            "designation": "Type of motion",
            "hasNestedStructure": false,
            "value": "rotating",
          },
          Object {
            "designation": "Speed (n_i)",
            "hasNestedStructure": false,
            "unit": "1/min",
            "value": "500.00",
          },
          Object {
            "designation": "Mean operating temperature (T)",
            "hasNestedStructure": false,
            "unit": "°C",
            "value": "50",
          },
          Object {
            "designation": "Radial load (Fr)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "500.0",
          },
          Object {
            "designation": "Axial load (Fa)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "500.0",
          },
        ],
        "title": "Loadcase 3",
      },
      Object {
        "hasNestedStructure": false,
        "subItems": Array [
          Object {
            "designation": "Time portion (q)",
            "hasNestedStructure": false,
            "unit": "%",
            "value": "10.000",
          },
          Object {
            "designation": "Type of motion",
            "hasNestedStructure": false,
            "value": "rotating",
          },
          Object {
            "designation": "Speed (n_i)",
            "hasNestedStructure": false,
            "unit": "1/min",
            "value": "500.00",
          },
          Object {
            "designation": "Mean operating temperature (T)",
            "hasNestedStructure": false,
            "unit": "°C",
            "value": "50",
          },
          Object {
            "designation": "Radial load (Fr)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "500.0",
          },
          Object {
            "designation": "Axial load (Fa)",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "500.0",
          },
        ],
        "title": "Loadcase 4",
      },
    ],
    "title": "Load",
    "titleID": "STRING_OUTP_INPUT_DATA_FOR_ALL_LOADCASES",
  },
]
`;

exports[`Calculation Result Selector full result getSelectedCalculations should return the selected calculations 1`] = `
Array [
  Object {
    "disabled": true,
    "label": "calculationTypes.co2ForProduction",
    "name": "emission",
    "resultAvailable": true,
    "selected": true,
    "svgIcon": "co2",
    "visible": true,
  },
  Object {
    "disabled": false,
    "icon": "compress",
    "label": "calculationTypes.frictionalPowerloss",
    "name": "frictionalPowerloss",
    "resultAvailable": true,
    "selected": true,
    "visible": true,
  },
]
`;
